pkg/utils/anonymize/unstructured.go

```diff
@@ -0,0 +1,22 @@
+package anonymize
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// UnstructuredNestedStringField anonymizes the nested field in the unstructured object
+// or returns error if it is not possible
+func UnstructuredNestedStringField(data map[string]interface{}, fields ...string) error {
+	value, found, err := unstructured.NestedFieldCopy(data, fields...)
+	if err != nil {
+		return err
+	}
+	if !found {
+		return fmt.Errorf("unable to find field '%v'", fields)
+	}
+
+	valueStr, _ := value.(string)
+	return unstructured.SetNestedField(data, String(valueStr), fields...)
+}
```

Comment:
I was doing something very similar in `pkg/utils/unstructerd_wrappers.go` (but without  anonymization/setting the field) so maybe we could move it there. WDYT?

Reply:
I think it's relates more to anonymize package than to utils.